src/axom/primal/operators/detail/evaluate_integral_impl.hpp

```diff
@@ -18,101 +18,137 @@
+          typename T,
+          int NDIMS,
+          typename RetType = std::invoke_result_t<Lambda, BezierCurve<T, NDIMS>::PointType>>
+inline RetType evaluate_line_integral_component(const BezierCurve<T, NDIMS>& c,
+                                                Lambda&& integrand,
+                                                const int npts)
 {
   const axom::numerics::QuadratureRule& quad = axom::numerics::get_gauss_legendre(npts);
 
   // Store/compute quadrature result
-  double full_quadrature = 0.0;
+  RetType full_quadrature = RetType {};
   for(int q = 0; q < npts; q++)
   {
     // Get intermediate quadrature point
     //  at which to evaluate tangent vector
     auto x_q = c.evaluate(quad.node(q));
     auto dx_q = c.dt(quad.node(q));
 
-    full_quadrature += quad.weight(q) * scalar_integrand(x_q) * dx_q.norm();
+    full_quadrature += quad.weight(q) * integrand(x_q) * dx_q.norm();
   }
 
   return full_quadrature;
 }
 
 /*!
- * \brief Evaluate a scalar field line integral on a single NURBS curve.
+ * \brief Evaluate a line integral on a single NURBS curve.
  *
  * Decompose the NURBS curve into Bezier segments, then sum the integral on each
  *
- * \tparam Lambda A callable type taking an axom::Point type returning a numeric type
+ * \tparam Lambda A callable type taking an CurveType's PointType and returning an integrable type
+ * \tparam RetType The return type of Lambda, which must support addition and scalar multiplication
  * \param [in] n The NURBS curve object
- * \param [in] scalar_integrand the lambda function representing the integrand. 
+ * \param [in] integrand the lambda function representing the integrand. 
  * \param [in] npts The number of quadrature points in the rule
  * \return the value of the integral
  */
-template <typename Lambda, typename T, int NDIMS>
-inline double evaluate_scalar_line_integral_component(const primal::NURBSCurve<T, NDIMS>& n,
-                                                      Lambda&& scalar_integrand,
-                                                      const int npts)
+template <typename Lambda,
+          typename T,
+          int NDIMS,
+          typename RetType = std::invoke_result_t<Lambda, NURBSCurve<T, NDIMS>::PointType>>
+inline RetType evaluate_line_integral_component(const NURBSCurve<T, NDIMS>& n,
+                                                Lambda&& integrand,
+                                                const int npts)
 {
   const auto beziers = n.extractBezier();
-  double total_integral = 0.0;
+  RetType total_integral = RetType {};
   for(int i = 0; i < beziers.size(); ++i)
   {
     total_integral +=
-      detail::evaluate_scalar_line_integral_component(beziers[i],
-                                                      std::forward<Lambda>(scalar_integrand),
-                                                      npts);
+      detail::evaluate_line_integral_component(beziers[i], std::forward<Lambda>(integrand), npts);
   }
   return total_integral;
```

Comment:
Similar suggestion to another above about using range-for instead of explicit loop

Would the following work?
```cpp
  RetType total_integral = RetType {};
  for(const auto& bez : n.extractBeziers())
  {
    total_integral += detail::evaluate_line_integral_component(bez, std::forward<Lambda>(integrand), npts);
  }
  return total_integral;
```

Feel free to ignore, but it might be more efficient since the Bezier curves are shorter lived.
Then again, it might be a wash.